homeassistant/components/switch/freebox.py

```diff
@@ -0,0 +1,94 @@
+"""
+Support for Freebox devices (Freebox v6 and Freebox mini 4K).
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/switch.freebox/
+"""
+import logging
+
+from homeassistant.components.freebox import DATA_FREEBOX
+from homeassistant.components.switch import SwitchDevice
+
+DEPENDENCIES = ['freebox']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up the sensors."""
+    fbx = hass.data[DATA_FREEBOX]
+    fbx_perms = FbxPerms(fbx, hass)
```

Comment:
Each platform will still has its own permission instance. If we want to avoid that, we should save it in hass.data and check there before creating a the permission instance.

Reply:
Or maybe better, create the permission instance in the component instead and save it in hass.data.